libs/shared/data/src/lib/form-builder.interface.ts

```diff
@@ -0,0 +1,11 @@
+import { FormlyFieldConfig, FormlyFormOptions } from '@ngx-formly/core';
+
+export interface IFormData {
+  [key: string]: any;
+}
+
+export type IFormField = FormlyFieldConfig;
+
+export type IFormFields = Array<IFormField>;
```

Comment:
redundant, just do IFormField[] where its needed (prefer to be explicit as its more readable IMO)